src/sys/time.rs

```diff
@@ -1,24 +1,24 @@
 use std::{fmt, ops};
-use libc::{time_t, suseconds_t};
-
-#[repr(C)]
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug)]
-pub struct TimeVal {
-    pub tv_sec: time_t,
-    pub tv_usec: suseconds_t,
-}
+use libc::{time_t, suseconds_t, timeval};
 
 const MICROS_PER_SEC: i64 = 1_000_000;
 const SECS_PER_MINUTE: i64 = 60;
 const SECS_PER_HOUR: i64 = 3600;
 
-#[cfg(target_pointer_width = "64")]
-const MAX_SECONDS: i64 = (::std::i64::MAX / MICROS_PER_SEC) - 1;
+#[derive(Clone, Copy)]
+pub struct TimeVal(pub timeval);
 
-#[cfg(target_pointer_width = "32")]
-const MAX_SECONDS: i64 = ::std::isize::MAX as i64;
+impl AsRef<timeval> for TimeVal {
+    fn as_ref(&self) -> &timeval {
+        &self.0
+    }
+}
 
-const MIN_SECONDS: i64 = -MAX_SECONDS;
+impl AsMut<timeval> for TimeVal {
+    fn as_mut(&mut self) -> &mut timeval {
+        &mut self.0
+    }
+}
```

Comment:
oh I like having `AsRef` and `AsMut` impls for newtypes. making a mental note to go back and look at what exists already...